src/backend/common/half.hpp

```diff
@@ -609,36 +815,25 @@ class alignas(2) half {
 
    public:
     half() = default;
-    CONSTEXPR_DH explicit half(double value) noexcept
-#ifdef __CUDA_ARCH__
-        : data_(__float2half(value)) {
-#else
-        : data_(float2half<std::round_to_nearest>(value)) {
-#endif
-    }
 
-    CONSTEXPR_DH explicit half(float value) noexcept
-#ifdef __CUDA_ARCH__
-        : data_(__float2half(value)) {
-#else
-        : data_(float2half<std::round_to_nearest>(value)) {
-#endif
+    /// Constructor.
+    /// \param bits binary representation to set half to
+    CONSTEXPR_DH half(internal::binary_t, uint16_t bits) noexcept : data_() {
+        memcpy(&data_, &bits, sizeof(uint16_t));
     }
 
-#ifndef __CUDACC_RTC__
-    template<typename T,
-        typename std::enable_if_t<std::is_unsigned<T>::value>* = nullptr>
-    CONSTEXPR_DH explicit half(T value) noexcept
-        : data_(int2half<std::round_to_nearest, false>(value)) {}
+    CONSTEXPR_DH explicit half(double value) noexcept
+        : data_(float2half<double>(value)) {}
```

Comment:
I assume, we don't have a doubl2half and hence we have float2half here. Shouldn't we just add `static_cast<float>(value)` too ?